modules/publication/ajax/FileUpload.php

```diff
@@ -0,0 +1,908 @@
+        'WHERE Name = :n AND Email = :e',
+        array(
+         'n' => $leadInvest,
+         'e' => $leadInvestEmail,
+        )
+    );
+    if (empty($leadInvID)) {
+        $db->insert(
+            'publication_collaborator',
+            array(
+             'Name'  => $leadInvest,
+             'Email' => $leadInvestEmail,
+            )
+        );
+
+        $leadInvID = $db->getLastInsertId();
+    }
+    if (!isset($desc, $leadInvest, $leadInvestEmail)) {
+        showPublicationError('A mandatory field is missing!', 400);
+    }
+    // INSERT INTO publication ...
+    $uid   = $user->getId();
+    $today = date('Y-m-d');
+    // insert the titleRaw to avoid double escaping
+    $fields = array(
+               'UserID'             => $uid,
+               'Title'              => $titleRaw,
+               'Description'        => $desc,
+               'LeadInvestigatorID' => $leadInvID,
+               'DateProposed'       => $today,
+              );
+
+    $db->insert('publication', $fields);
+    $pubID = $db->getLastInsertId();
+
+    try {
+        // process files
+        processFiles($pubID);
+        // INSERT INTO publication_collaborator
+        insertCollaborators($pubID);
+        // INSERT INTO publication_users_edit_perm_rel
+        insertEditors($pubID);
+        // INSERT INTO publication_keyword
+        insertKeywords($pubID);
+        // INSERT INTO publication_parameter_type_rel
+        insertVOIs($pubID);
+    } catch (Exception $e) {
+        cleanup($pubID);
+        showPublicationError($e->getMessage(), $e->getCode());
+    }
+
+    notify($pubID, 'submission');
+}
+
+/**
+ * Function for storing files and inserting file meta data into database
+ *
+ * @param int $pubID ID of Publication in publication table
+ *
+ * @return void
+ */
+function processFiles($pubID) : void
+{
+    if (empty($_FILES)) {
+        return;
+    }
+    $db     = \Database::singleton();
+    $config = \NDB_Config::singleton();
+
+    $publicationPath = $config->getSetting('publication_uploads');
+
+    if (!isset($publicationPath)) {
+        throw new LorisException(
+            "Error! Publication path is not set in Loris Settings!",
+            500
+        );
+    }
+
+    if (!file_exists($publicationPath)) {
+        throw new LorisException(
+            "Error! The upload folder '$publicationPath' does not exist!",
+            500
+        );
+    }
+
+    foreach ($_FILES as $name => $values) {
+        $fileName = preg_replace('/\s/', '_', $values["name"]);
+        if (file_exists($publicationPath . $fileName)) {
+            throw new LorisException("File $fileName already exists!", 409);
+        }
+        $extension = pathinfo($fileName)['extension'];
+        $index     = preg_split('/_/', $name)[1];
+
+        if (!isset($extension)) {
+            throw new LorisException(
+                "Please make sure your file has a valid extension: " .
+                $values['name'],
+                400
```

Comment:
The error code in PHP Exceptions (and LorisExceptions by extension) is not the same as an HTTP response code. Throwing a LorisException will always give a 500 response code. If you want to give a `400` HTTP Response code with some debug info, do:

```php
http_response_code(400);
echo $some_helpful_err_msg;
``